src/imgtools/utils/nnunet.py

```diff
@@ -1,103 +1,176 @@
-    json_dict['numTest'] = len(test_identifiers)
-    json_dict['training'] = [
-        {'image': "./imagesTr/%s.nii.gz" % i, "label": "./labelsTr/%s.nii.gz" % i} for i
-        in
-        train_identifiers]
-    json_dict['test'] = ["./imagesTs/%s.nii.gz" % i for i in test_identifiers]
-
-    if not output_file.endswith("dataset.json"):
-        print("WARNING: output file name is not dataset.json! This may be intentional or not. You decide. "
-              "Proceeding anyways...")
-    save_json(json_dict, os.path.join(output_file), sort_keys=sort_keys)
+    Generates a dataset.json file in the output folder
+
+    channel_names:
+        Channel names must map the index to the name of the channel, example:
+        {
+            0: 'T1',
+            1: 'CT'
+        }
+        Note that the channel names may influence the normalization scheme!! Learn more in the documentation.
+
+    labels:
+        This will tell nnU-Net what labels to expect. Important: This will also determine whether you use region-based training or not.
+        Example regular labels:
+        {
+            'background': 0,
+            'left atrium': 1,
+            'some other label': 2
+        }
+        Example region-based training:
+        {
+            'background': 0,
+            'whole tumor': (1, 2, 3),
+            'tumor core': (2, 3),
+            'enhancing tumor': 3
+        }
+
+        Remember that nnU-Net expects consecutive values for labels! nnU-Net also expects 0 to be background!
+
+    num_training_cases: is used to double check all cases are there!
+
+    file_ending: needed for finding the files correctly. IMPORTANT! File endings must match between images and
+    segmentations!
+
+    dataset_name, reference, release, license, description: self-explanatory and not used by nnU-Net. Just for
+    completeness and as a reminder that these would be great!
+
+    overwrite_image_reader_writer: If you need a special IO class for your dataset you can derive it from
+    BaseReaderWriter, place it into nnunet.imageio and reference it here by name
+
+    kwargs: whatever you put here will be placed in the dataset.json as well
+
+    """
+
+    has_regions: bool = any([isinstance(i, (tuple, list)) and len(i) > 1 for i in labels.values()])
+    if has_regions:
+        assert regions_class_order is not None, f"You have defined regions but regions_class_order is not set. " \
+                                                f"You need that."
+    # channel names need strings as keys
+    keys = list(channel_names.keys())
+    for k in keys:
+        if not isinstance(k, str):
+            channel_names[str(k)] = channel_names[k]
+            del channel_names[k]
+
+    # labels need ints as values
+    for l in labels.keys():
+        value = labels[l]
+        if isinstance(value, (tuple, list)):
+            value = tuple([int(i) for i in value])
+            labels[l] = value
+        else:
+            labels[l] = int(labels[l])
+
```

Comment:
this is a little confusing, and letting the function be called with multiple possible types and then handling it can lead to bugs 

we are also dealing with a similar issue in https://github.com/bhklab/med-imagetools/blob/22d8fb77f1d2c47d26967b1578e9be41d829dbb1/src/imgtools/modules/structureset.py#L167-L186 

becomes really hard to follow along and predict whats going to happen 

better off enforcing it to be the way you want and change the function call in autopipeline to pass labels in the way we want it to

Reply:
Yeah I agree. The function is from nnunet and they want to do this. Ima just remove it